File: 20-valid-parentheses/valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
          stack<char> stakk;
        int i = 0;
        while (i < s.size()) {
            if (s[i] == '(' || s[i] == '[' || s[i] == '{') {
                stakk.push(s[i]);
                i++;
            } else if (!stakk.empty() && s[i] == ')' && stakk.top() == '(') {
                stakk.pop();
                i++;
            } else if (!stakk.empty() && s[i] == ']' && stakk.top() == '[') {
                stakk.pop();
                i++;
            } else if (!stakk.empty() && s[i] == '}' && stakk.top() == '{') {
                stakk.pop();
                i++;
            } else {
                return false;
            }
        }
        return stakk.empty();
    }
        
    
};

  

File: 20-valid-parentheses/valid-parentheses.java
class Solution {
    public boolean isValid(String s) {
        int n=s.length();
        Stack<Character>st=new Stack<>();
        for(int i=0;i<n;i++){
            char c=s.charAt(i);
            if(c=='('||c=='['||c=='{'){
                st.push(c);
            }else{
                if(c==')'){
                    if(st.isEmpty()||st.pop()!='('){
                        return false;
                    }

                }
                else if(c==']'){
                    if(st.isEmpty()||st.pop()!='['){
                        return false;
                    }
                }
                else if(c=='}'){
                    if(st.isEmpty()||st.pop()!='{'){
                        return false;
                    }
                }
            }
        }
        return st.isEmpty();
        
    }
}